genesyscloud/user/resource_genesyscloud_user_test.go

```diff
@@ -0,0 +1,1307 @@
+					email1,
+					userName,
+					"routing_utilization = []",
+				),
+				Check: resource.ComposeTestCheckFunc(
+					validateUserUtilizationLevel(resourceName+"."+userResource1, "Organization"),
+					resource.TestCheckNoResourceAttr(resourceName+"."+userResource1, "routing_utilization.%"),
+				),
+			},
+		},
+		CheckDestroy: testVerifyUsersDestroyed,
+	})
+}
+
+func TestAccResourceUserRestore(t *testing.T) {
+	t.Parallel()
+	var (
+		userResource1 = "test-user"
+		email1        = "terraform-" + uuid.NewString() + "@user.com"
+		userName1     = "Terraform Restore1"
+		//userName2     = "Terraform Restore2" //TODO
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { util.TestAccPreCheck(t) },
+		ProviderFactories: provider.GetProviderFactories(providerResources, providerDataSources),
+		Steps: []resource.TestStep{
+			{
+				// Create a basic user
+				Config: GenerateBasicUserResource(
+					userResource1,
+					email1,
+					userName1,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName+"."+userResource1, "email", email1),
+					resource.TestCheckResourceAttr(resourceName+"."+userResource1, "name", userName1),
+				),
+			},
+			{
+				Config: GenerateBasicUserResource(
+					userResource1,
+					email1,
+					userName1,
+				),
+				Destroy: true, // Delete the user
+				Check:   testVerifyUsersDestroyed,
+			},
+			/*{ //TODO
+				// Restore the same user email but set a different name
+				Config: GenerateBasicUserResource(
+					userResource1,
+					email1,
+					userName2,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName+"."+userResource1, "email", email1),
+					resource.TestCheckResourceAttr(resourceName+"."+userResource1, "name", userName2),
+				),
+			},*/
+		},
+		CheckDestroy: testVerifyUsersDestroyed,
+	})
+}
+
+func TestAccResourceUserCreateWhenDestroyed(t *testing.T) {
+	t.Parallel()
+	var (
+		userResource1 = "test-user"
+		email1        = "terraform-" + uuid.NewString() + "@user.com"
+		userName1     = "Terraform Existing"
+		//userName2     = "Terraform Create" //TODO
+		//stateActive   = "active"
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { util.TestAccPreCheck(t) },
+		ProviderFactories: provider.GetProviderFactories(providerResources, providerDataSources),
+		Steps: []resource.TestStep{
+			{
+				// Create a basic user
+				Config: GenerateBasicUserResource(
+					userResource1,
+					email1,
+					userName1,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName+"."+userResource1, "email", email1),
+					resource.TestCheckResourceAttr(resourceName+"."+userResource1, "name", userName1),
+				),
+			},
+			{
+				Config: GenerateBasicUserResource(
+					userResource1,
+					email1,
+					userName1,
+				),
+				Destroy: true, // Delete the user
+				Check:   testVerifyUsersDestroyed,
+			},
+			/*{ //TODO
```

Comment:
I am getting error in this testing step, that needs to be fixed.

Reply:
Any update on this? This test is passing for me as it is on the dev branch. Strange that it would not be working here

Reply:
not yet, this week i will start working on this again and will update the code. good to know that i works for you in the dev branch let me also try that from myside as well.

Reply:
I fixed it will commit the changes